// react-materialize/lib/Col.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _constants = require('./constants');

var _constants2 = _interopRequireDefault(_constants);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Col = function Col(_ref) {
  var children = _ref.children;
  var className = _ref.className;
  var _ref$node = _ref.node;
  var node = _ref$node === undefined ? 'div' : _ref$node;
  var offset = _ref.offset;

  var props = _objectWithoutProperties(_ref, ['children', 'className', 'node', 'offset']);

  var Child = node;
  var classes = { col: true };
  _constants2.default.SIZES.forEach(function (s) {
    if (props[s]) {
      classes[s + props[s]] = true;
    }
  });

  if (offset) {
    offset.split(' ').forEach(function (off) {
      classes['offset-' + off] = true;
    });
  }

  return _react2.default.createElement(
    Child,
    { className: (0, _classnames2.default)(classes, className) },
    children
  );
};

Col.propTypes = {
  children: _react2.default.PropTypes.node,
  className: _react2.default.PropTypes.string,
  /**
   * Columns for large size screens
   */
  l: _react2.default.PropTypes.number,
  /**
   * Columns for middle size screens
   */
  m: _react2.default.PropTypes.number,
  /**
   * The node to be used for the column
   * @default div
   */
  node: _react2.default.PropTypes.node,
  /**
   * To offset, simply add s2 to the class where s signifies the screen
   * class-prefix (s = small, m = medium, l = large) and the number after
   * is the number of columns you want to offset by.
   */
  offset: _react2.default.PropTypes.string,
  /**
   * Columns for small size screens
   */
  s: _react2.default.PropTypes.number
};

exports.default = Col;
